### ISSUE TITLE
link and link_tree in manifest file

### ISSUE THREAD

**Issue by DanCouper** [rails/sprockets#161]
This may well be a misunderstanding on my part, or it may be just because the API is in flux, but I'm having an issue with Sprockets 4 locating assets.

I've an asset structure that looks like this:

![screen shot 2015-11-19 at 11 54 23](https://cloud.githubusercontent.com/assets/573694/11270149/541bca32-8eb4-11e5-8e08-15f469a4858d.png)

The `_variables.scss`, `_colours.scss` and `_mixins.scss` are `@import`ed at the top of the application file, and are referenced in the remaining files.  `manifest.js` looks basically like this at the minute, as I can't get it to work any other way despite what the Sprockets error messages helpfully tell me are the other options:

```
// JS and CSS bundles
//= link_tree ../javascripts .js
//= link_tree ../stylesheets .css
```

What this means is that I have to manually `@import` the variables *etc* into every subsequent file, rather than just declaring them once. I assumed that I could use `link` rather than `link_tree`, as specified in the error messages and in `UPGRADING.md`, but changing to `//= link ../stylesheets/application.css` gives me the error `couldn't find file '../../stylesheets/application.css' under '/Users/danielcouper/Projects/my_project/app/assets/config'`. `link` seems to behave differently to `link_tree`, with the relative path failing in the former - it's as if it's completely ignoring the filepath. This is contrary to the error message I get if I move the manifest file out of `assets/config/`:

```
Asset `application.css` was not declared to be precompiled in production.
Declare links to your assets in `assets/config/manifest.js`.
Examples:
`//= link ../javascripts/application.js`
`//= link_directory ../javascripts .js`
`//= link_directory ../stylesheets .css`
`//= link_tree ../javascripts .js`
`//= link_tree ../images`
and restart your server
```
The first option, the one I need to implement, seems to flat out not work. I'm a little bit stumped and I'm not sure if this is a bug or whether I'm doing something wrong here.

**Comment by rafaelfranca** [comment]
How many css files you want to sprockets to generate? If you only need `application.css` you should be using `link` instead of `link_tree`. Try using `link` as the following. `//= link application.css`.

**Comment by DanCouper** [comment]
Ah, right, I see that fixes my issue. Yes, `//= link application.css` and `//= link application.js` do work (thank you). However, I was following the suggestion of the error message (*ie* `//= link ../javascripts/application.js`), which did not work. It was that that was causing me confusion; Sprockets simply would not resolve that [explicit] filepath correctly, it gave me an error stating that application.css could not be found under `my/app/path/app/assets/config`.

I had to blow away and reinstall all my gems yesterday, and I now cannot now replicate this error, so hopefully was just something other than Sprockets. I'll keep an eye on it anyway and see if the issue crops up again

**Comment by hallaathrad** [comment]
Hey! Having the same issue and getting past the error message doing the same (not following the example, but linking the file directly). The issue now is the scss files are not being processed and shown... rendering the page entirely unstyled.

Being entirely new at the Rails thing (JavaScript front-end here), I'm sort of at loss on how to solve this. Help, please?

**Comment by rafaelfranca** [comment]
@hallaathrad could you explain exactly what are you trying to do and that is not working?

**Comment by hallaathrad** [comment]
Thanks @rafaelfranca
So, long story short, I've just joined a project using legacy gems which I've been trying to update so I can set the front-end parts of it in order... finally just managed to get rails serving again after dealing with dependency mayhem, and as said before: no previous experience at rails... I've been a node/JS guy until now.

The site was rendering "fine" until I started updating gems, due to needing a newer jQuery version.
Just solved the "create a manifest.js file" resulting issue from updating Sprockets, and then using your advice from your [comment ☝️](https://github.com/rails/sprockets/issues/161#issuecomment-159023538).

But now the page, though it's running and being displayed, it's not loading any of the scss required by `application.css`. I'm guessing they're not being compiled into css, nor do I see them being injected into the rendered html. And of course, I'm at loss on what to do or where to turn it on.
